File: swift/structTest2.swift
struct Person{
    let name:String
    let age:Int
    var fruits:[String]
    
    func say(){
        print(self.name, self.age, self.fruits)
    }
    
    mutating func have(fruit:String){
        fruits.append(fruit)
    }
    
    mutating func remove() -> String{
        return fruits.removeLast()
    }
}

func test(){
    var a = Person(name:"Stanley", age:20, fruits:["apple"])
    a.have(fruit:"banana")
    a.have(fruit:"orange")
    print(a.remove())
    a.say()
}

File: rust/test2/src/structTest2.rs

struct Person {
	name: String,
	age: i32
}

impl Person {
	fn new(name:String, age:i32) -> Person{
		Person{name, age}
	}
	fn say(&self){
		println!("{} {}", self.name, self.age);
	}
}

pub fn test(){
	let p =Person::new(String::from("Stanley"), 20);
	p.say();
}